src/strategies/MangroveOrder.sol

```diff
@@ -120,17 +141,17 @@ contract MangroveOrder is Forwarder, IOrderLogic {
     // * (NAT_THIS+`msg.value`+`res.bounty`, OUT_THIS+`res.takerGot`, IN_THIS+`tko.takerGives`-`res.takerGave`)
 
     bool isComplete = checkCompleteness(tko, res);
-    // when `!restingOrder` this implements FOK. When `restingOrder` the `postRestingOrder` function reverts if resting order fails to be posted amd `partialFillNotAllowed`.
-    // therefore we require `partialFillNotAllowed => (isComplete \/ restingOrder)`
-    require(!tko.partialFillNotAllowed || isComplete || tko.restingOrder, "mgvOrder/mo/noPartialFill");
+    // when `!restingOrder` this implements FOK. When `restingOrder` the `postRestingOrder` function reverts if resting order fails to be posted and `fillOrKill`.
+    // therefore we require `fillOrKill => (isComplete \/ restingOrder)`
+    require(!tko.fillOrKill || isComplete || tko.restingOrder, "mgvOrder/partialFill");
 
     // sending inbound tokens to `msg.sender`'s reserve and sending back remaining outbound tokens
     if (res.takerGot > 0) {
-      router().push(tko.outbound_tkn, callerReserve, res.takerGot);
+      require(router().push(tko.outbound_tkn, callerReserve, res.takerGot) == res.takerGot, "mgvOrder/pushFailed");
```

Comment:
`push` returns the amount that was pushed to reserve. In the specific case of SimpleRouter it is either all or nothing. But we check the returned amount in case strat builder changes router policy